packages/pwa-kit-dev/src/configs/webpack/plugins.js

```diff
@@ -58,20 +64,30 @@ export const createModuleReplacementPlugin = (projectDir) => {
         })
     })
 
+    // Return a new `webpack.NormalModuleReplacementPlugin` instance
     return new webpack.NormalModuleReplacementPlugin(/.*/, (resource) => {
-        const resolved = path.resolve(resource.context, resource.request)
+        // We only want to replace resources that are requested from the SDK
+        if (resource.context.includes('pwa-kit-react-sdk')) {
+            // Resolve the full path of the resource
+            const resolved = path.resolve(resource.context, resource.request)
 
-        const replacement = replacements.find(({path}) => resolved.match(path))
+            // Find the replacement for the resolved path from the overridables list
+            const replacement = replacements.find(({path}) => resolved.match(path))
 
-        const sdkPaths = [
-            path.join('packages', 'pwa-kit-react-sdk'),
-            path.join('node_modules', 'pwa-kit-react-sdk')
-        ]
+            if (replacement) {
+                // Check if the resource was requested from 'packages/pwa-kit-react-sdk' or 'node_modules/pwa-kit-react-sdk'
+                const sdkPaths = [
+                    path.join('packages', 'pwa-kit-react-sdk'),
+                    path.join('node_modules', 'pwa-kit-react-sdk')
+                ]
 
-        const requestedFromSDK = sdkPaths.some((p) => resource.context.includes(p))
+                const requestedFromSDK = sdkPaths.some((p) => resource.context.includes(p))
 
-        if (requestedFromSDK && replacement) {
-            resource.request = replacement.newPath
+                // If the resource was requested from SDK, replace the resource request with the replacement path
+                if (requestedFromSDK) {
```

Comment:
How about moving this check earlier in the logic? I think conceptually it goes together with the first `if` condition.

So I'm thinking something like this:
```js
if (resource.context.includes('pwa-kit-react-sdk') && sdkPaths.some((p) => resource.context.includes(p))) {
  ...
}
```
That `(A && B)` condition still follows your intention of short-circuiting when `resource.context` does not include "pwa-kit-react-sdk".